main/src/ca/uwaterloo/flix/tools/pkg/github/GitHub.scala

```diff
@@ -78,11 +78,24 @@ object GitHub {
     * Publish a new release the given project.
     */
   def publishRelease(project: Project, version: SemVer, apiKey: String): Result[Unit, ReleaseError] = {
-    val content: JValue = ("tag_name" -> s"v$version") ~ ("name" -> s"v$version")
+    for (
+      url <- createDraftRelease(project, version, apiKey);
+      _ <- markReleaseReady(project, version, url, apiKey)
+    ) yield Ok(())
+  }
+
+  /**
+    * Create a new release marked as a draft, meaning that it is not publicly visible.
+    * The release will not contain any assets (apart from the default zips of the source code).
+    *
+    * Returns the URL of the release if successful.
+    */
+  private def createDraftRelease(project: Project, version: SemVer, apiKey: String): Result[URL, ReleaseError] = {
```

Comment:
Is this required? Why the split into draft/non-draft?

We want to just create a non-draft release immediately.

Reply:
Since uploading the artifacts has to happen as a seperate fallible step, I figured it was just better practice to not make the release visible before it's entirely done. It's more complex yes, but I see it as the only sound way to do it.

Reply:
So whats the workflow, and is it hidden from the programmer?

Reply:
Yes, it's:
- Create draft
- Upload all assets
- Make public

It should all happen immediately unless an error is encountered, in which case it just leaves an unfinished draft that is only visible to yourself and doesn't stop you from creating a new release with the same name.

Reply:
Okay. Lets try that 👍